src/libraries/Microsoft.Extensions.DependencyInjection.Specification.Tests/src/ActivatorUtilitiesTests.cs

```diff
@@ -350,6 +378,39 @@ public void GetServiceOrCreateInstanceUnregisteredService()
             }
         }
 
+        [Fact]
+        public void CreateInstance_BadlyConfiguredIServiceProviderIsService_ProperlyCreatesUnambiguousInstance()
+        {
+            if (SupportsIServiceProviderIsService)
+            {
+                return;
+            }
+
+            var serviceCollection = new TestServiceCollection()
+                .AddScoped<FakeService1>();
+            var fakeIServiceProviderIsService = new FakeIServiceProviderIsService();
+
+            serviceCollection.AddScoped<IServiceProviderIsService>((p) => (IServiceProviderIsService)fakeIServiceProviderIsService);
+            var serviceProvider = CreateServiceProvider(serviceCollection);
+
+            var instance = ActivatorUtilities.CreateInstance<FakeService2>(serviceProvider);
+            Assert.NotNull(instance);
+            Assert.True(fakeIServiceProviderIsService.Called);
+        }
+
+        [Theory]
+        [MemberData(nameof(CreateInstanceFuncs))]
+        public void CreateInstance_NullInstance_HandlesBadInputWithInvalidOperationException(CreateInstanceFunc createFunc)
```

Comment:
This test should also be moved to DI.Tests\ActivatorUtilitiesTests.cs. It is testing ActivatorUtilities, not the external DI implementations.